modules/45_carbonprice/diffLin2Lin/datainput.gms

```diff
@@ -27,36 +27,38 @@ if(cm_co2_tax_spread eq 10,
 p45_phasein_2025ratio(regi)$(p45_gdppcap2015_PPP(regi) gt 5   and p45_gdppcap2015_PPP(regi) le 10) = 0.2; !! OAS
-p45_phasein_2025ratio(regi)$(p45_gdppcap2015_PPP(regi) gt 10  and p45_gdppcap2015_PPP(regi) le 15) = 0.4; !! CHA, REF, LAM
-p45_phasein_2025ratio(regi)$(p45_gdppcap2015_PPP(regi) gt 15  and p45_gdppcap2015_PPP(regi) le 20) = 0.6; !! MEA, NEU
-p45_phasein_2025ratio(regi)$(p45_gdppcap2015_PPP(regi) gt 20) = 1; !! EUR, JPN, USA, CAZ
+p45_phasein_2025ratio(regi)$(p45_gdppcap2015_PPP(regi) gt 10  and p45_gdppcap2015_PPP(regi) le 15) = 0.4; !! CHA, LAM, MEA
+p45_phasein_2025ratio(regi)$(p45_gdppcap2015_PPP(regi) gt 15  and p45_gdppcap2015_PPP(regi) le 20) = 0.6; !! REF
+p45_phasein_2025ratio(regi)$(p45_gdppcap2015_PPP(regi) gt 20) = 1; !! EUR, JPN, USA, CAZ, NEU
 );
 display p45_phasein_2025ratio;
 
 
 *** for the current implementation, use the following trajectory for rich countries:
-*** price increases linearly from 0 in 2020 (warning: quick fix for more near-term realism. To be changed to NPI fixing) until the pkBudgYr, then increases with c_taxCO2inc_after_peakBudgYr
-if(cm_co2_tax_2020 lt 0,
-  abort "please choose a valid cm_co2_tax_2020"
-elseif cm_co2_tax_2020 ge 0,
+*** price increases linearly from 25$/t CO2eq in 2020 until peak budget year (if cm_iterative_target_adj = 6|7|9) or until end of century (otherwise), 
+*** then increases with c_taxCO2inc_after_peakBudgYr
+if(cm_co2_tax_startyear le 25,
+  abort "please choose a valid cm_co2_tax_startyear of more than 25$/t CO2eq"
+else
 *** convert tax value from $/t CO2eq to T$/GtC
-  p45_CO2priceTrajDeveloped("2040")= 3 * cm_co2_tax_2020 * sm_DptCO2_2_TDpGtC;  !! shifted to 2040 to make sure that even in delay scenarios the fixpoint of the linear price path is inside the "t" range, otherwise the CO2 prices from reference run may be overwritten
-*** The factor 3 is inherited from old fixing to 0 in 2010. Switch cm_co2_tax_2020 should be renamed or removed.
+  s45_co2_tax_2020      = 25 * sm_DptCO2_2_TDpGtC;
+  s45_co2_tax_startyear = cm_co2_tax_startyear * sm_DptCO2_2_TDpGtC;
 );
-
-
-
-p45_CO2priceTrajDeveloped(t)$(t.val ge 2020) = p45_CO2priceTrajDeveloped("2040")*( 1 + (t.val-2040) / 20); !! no CO2 price until 2020 and only change CO2 prices after ; 
-*** annual increase by 5% (=1/20) of the 2040 value corresponds to a linear increase from 0 in 2020 to the 2040 value
+*** define linear curve through the points (2020, s45_co2_tax_2020) and (cm_startyear, s45_co2_tax_startyear)
+p45_CO2priceTrajDeveloped(t) = s45_co2_tax_2020
+                                + (s45_co2_tax_startyear - s45_co2_tax_2020) / (cm_startyear - 2020) !! Yearly increase of CO2 price 
+                                  * (t.val - 2020) ;
+p45_CO2priceTrajDeveloped(t)$(t.val gt 2110) = p45_CO2priceTrajDeveloped("2110"); !! set taxes constant after 2110
+*** for peak budget runs (if cm_iterative_target_adj = 6|7|9), the adjustment of the CO2 price trajectory after the peak year is made in core/postsolve.gms
```

Comment:
This is formula is the core of the change. It also appears in the postsolve of this module.